src/map/unit.cpp

```diff
@@ -2784,10 +2812,12 @@ static int unit_attack_timer_sub(struct block_list* src, int tid, t_tick tick)
 			unit_stop_walking(src,1);
 
 		if(md) {
-			//First attack is always a normal attack
-			if(md->state.skillstate == MSS_ANGRY || md->state.skillstate == MSS_BERSERK) {
-				if (mobskill_use(md,tick,-1))
+			// Berserk skills can replace normal attacks except for the first attack (state not berserk yet)
+			if(md->state.skillstate == MSS_BERSERK) {
+				if (mobskill_use(md, tick, -1)) {
+					ud->attackabletime = tick + sstatus->adelay;
```

Comment:
Why is this not set inside the function already?

Reply:
Same as above, it's only set for Angry and Berserk skills.

Reply:
Then put that logic into the function please.
You can access it there via md->state.skillstate and make it totally clear and understandable there.

Reply:
Setting the delay is situational, I'll add a comment instead.

Reply:
Comment added.